test/integration/replicaset/replicaset_test.go

```diff
@@ -162,19 +173,247 @@ func rmSimpleSetup(t *testing.T) (*httptest.Server, framework.CloseFunc, clients
+	stopCh := make(chan struct{})
+	informers.Start(stopCh)
+	go rm.Run(5, stopCh)
+	return stopCh
+}
+
 // wait for the podInformer to observe the pods. Call this function before
 // running the RS controller to prevent the rc manager from creating new pods
 // rather than adopting the existing ones.
 func waitToObservePods(t *testing.T, podInformer cache.SharedIndexInformer, podNum int) {
-	if err := wait.Poll(10*time.Second, 60*time.Second, func() (bool, error) {
+	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
 		objects := podInformer.GetIndexer().List()
-		if len(objects) == podNum {
-			return true, nil
-		} else {
-			return false, nil
+		return len(objects) == podNum, nil
+	}); err != nil {
+		t.Fatalf("Error encountered when waiting for podInformer to observe the pods: %v", err)
+	}
+}
+
+func createRSsPods(t *testing.T, clientSet clientset.Interface, rss []*v1beta1.ReplicaSet, pods []*v1.Pod) ([]*v1beta1.ReplicaSet, []*v1.Pod) {
+	ns := rss[0].Namespace
+	rsClient := clientSet.Extensions().ReplicaSets(ns)
+	podClient := clientSet.Core().Pods(ns)
+	var createdRSs []*v1beta1.ReplicaSet
+	var createdPods []*v1.Pod
+	for _, rs := range rss {
+		createdRS, err := rsClient.Create(rs)
+		if err != nil {
+			t.Fatalf("Failed to create replica set %s: %v", rs.Name, err)
+		}
+		createdRSs = append(createdRSs, createdRS)
+	}
+	for _, pod := range pods {
+		createdPod, err := podClient.Create(pod)
+		if err != nil {
+			t.Fatalf("Failed to create pod %s: %v", pod.Name, err)
+		}
+		createdPods = append(createdPods, createdPod)
+	}
+
+	return createdRSs, createdPods
+}
+
+// Verify .Status.Replicas is equal to .Spec.Replicas
+func waitRSStable(t *testing.T, clientSet clientset.Interface, rs *v1beta1.ReplicaSet) {
+	rsClient := clientSet.Extensions().ReplicaSets(rs.Namespace)
+	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		newRS, err := rsClient.Get(rs.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return newRS.Status.Replicas == *rs.Spec.Replicas, nil
+	}); err != nil {
+		t.Fatalf("Failed to verify .Status.Replicas is equal to .Spec.Replicas for rs %s: %v", rs.Name, err)
+	}
+}
+
+// Update .Spec.Replicas to replicas and verify .Status.Replicas is changed accordingly
+func scaleRS(t *testing.T, c clientset.Interface, rs *v1beta1.ReplicaSet, replicas int32) {
+	rsClient := c.Extensions().ReplicaSets(rs.Namespace)
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		newRS, err := rsClient.Get(rs.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		*newRS.Spec.Replicas = replicas
+		rs, err = rsClient.Update(newRS)
+		return err
+	}); err != nil {
+		t.Fatalf("Failed to update .Spec.Replicas to %d for rs %s: %v", replicas, rs.Name, err)
+	}
+	waitRSStable(t, c, rs)
+}
+
+func updatePod(t *testing.T, podClient typedv1.PodInterface, pod *v1.Pod, updateFunc func(*v1.Pod)) {
```

Comment:
@enisoc `updatePod` and `updatePodStatus` differs by 2 lines of codes. It is possible to combine them together, but i prefer to have separate functions (one for spec update, one for status update). WDYT?